Grade: 45.00/50.00

1 Deduction(s):

--------------
#1: 5.00 points
Failing test: test_query_to_vector_unknown: 
        @points=5
        
Traceback (most recent call last):
  File "assignments-sol/a1/a1_test.py", line 82, in test_query_to_vector_unknown
    vec = idx.query_to_vector(['a', 'b', 'c', 'd'])
  File "students/balajisuiaji474/a1/searcher.py", line 150, in query_to_vector
    index[word] = log10(float(len(self.documents))/float(self.doc_freqs[word]))
KeyError: 'd'

source:
    def test_query_to_vector_unknown(self):
         """
         @points=5
         """
         idx = Index()
         idx.doc_freqs = {'a': 10, 'b': 100, 'c': 1000}
         idx.documents = [1] * 1000
         vec = idx.query_to_vector(['a', 'b', 'c', 'd'])
         self.assertAlmostEqual(vec['a'], 2.0, 1)
         self.assertAlmostEqual(vec['b'], 1.0, 1)
         self.assertAlmostEqual(vec['c'], 0.0, 1)
         self.assertFalse('d' in vec)
--------------

